src/component.ts

```diff
@@ -1,5 +1,28 @@
-import {Component, ProviderMap} from '@loopback/core';
+import {Component, CoreBindings, ProviderMap, inject} from '@loopback/core';
+import {Class, Model} from '@loopback/repository';
+import {AuditLog} from './models';
+import {AuditLog as TenantAuditLog} from './models/tenant-support';
+import {
+  IDisableTenantGuard,
+  TenantUtilitiesBindings,
+  TenantUtilitiesComponent,
+} from '@sourceloop/core';
+import {RestApplication} from '@loopback/rest';
 
 export class AuditLogComponent implements Component {
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE)
+    private readonly application: RestApplication,
+    @inject(TenantUtilitiesBindings.DisableTenantGuard, {optional: true})
+    private readonly disableTenantGuard?: IDisableTenantGuard,
+  ) {
+    this.application.component(TenantUtilitiesComponent);
```

Comment:
why not bind the component only if required ?

Reply:
because it is required for tenant guard, as tenant guard is called every time , it only skips its further flow if we provide config